Fix the JavaScriptCore build in some configurations

rr-codes opened the pull request:
#### b79359af12175bca7258e35b49c98208eca3cbf2
<pre>
Fix the JavaScriptCore build in some configurations
<a href="https://bugs.webkit.org/show_bug.cgi?id=300635">https://bugs.webkit.org/show_bug.cgi?id=300635</a>
<a href="https://rdar.apple.com/162411683">rdar://162411683</a>

Reviewed by Keith Miller.

First, some background:

1. The `asm` keyword is a supported keyword in C++, and GNU C. Notably however, it does not exist in non-GNU C standards.
Instead, the alternative keyword `__asm__` exists to serve the same purpose.

2. The bmalloc module map consists of a `pas` submodule, which participates in header fusion. This means that if a client
uses any of the headers in it, it implicitly imports all the other headers. This is expected and desired behavior.

3. The JavaScriptCore private module map defines a submodule &quot;PASReportCrashPrivate&quot; which contains the associated header,
and PASReportCrashPrivate.h transitively depends on at least one of the headers in the `bmalloc.pas` modules.

4. Within the bmalloc.pas module is pas_utils.h and pas_utils_prefix.h, both of which use the `asm` keyword.

As a result of all of these, the JavaScriptCore private module map depends on the existence of the `asm` keyword in all cases,
whereas before it was limited to the inclusion of specific headers. Consequently, any non-GNU C clients of the JavaScriptCore private
module will fail to compile because `asm` does not exist.

Fix by moving to `__asm__` which is available everywhere.

* Source/bmalloc/libpas/src/libpas/pas_utils.h:
(pas_assertion_failed):
(pas_reverse):
(pas_reverse64):
(pas_atomic_store_uint8):
(pas_compare_and_swap_uint8_weak):
(pas_compare_and_swap_uint8_strong):
(pas_compare_and_swap_uint16_weak):
(pas_compare_and_swap_uint32_weak):
(pas_compare_and_swap_uint32_strong):
(pas_compare_and_swap_uint64_weak):
(pas_compare_and_swap_uint64_strong):
(pas_store_store_fence):
(pas_opaque):
(pas_compare_and_swap_pair_weak):
(pas_compare_and_swap_pair_strong):
(pas_atomic_store_pair):
(pas_compare_ptr_opaque):
* Source/bmalloc/libpas/src/libpas/pas_utils_prefix.h:
(__pas_compiler_fence):
(__pas_fence):
(__pas_depend_impl):

Canonical link: <a href="https://commits.webkit.org/301429@main">https://commits.webkit.org/301429@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e8cceee4ed9e55479006767f793065998c0d7e22

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125943 "18 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45604 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36373 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132812 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77802 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/51df10ed-e06d-4fa0-aaed-5acbc5b734a6) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46288 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54162 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/95956 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64048 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d7f5a00a-23be-44e2-b220-7c9d1b7afc96) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128891 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37019 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112637 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76445 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/322f4ebb-414b-402c-bed5-5400b35f5ab7) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35927 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30822 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76282 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/118023 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106801 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31035 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135497 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/124450 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52727 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40458 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104433 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53176 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108848 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104153 "Found 1 new API test failure: WebKitGTK/TestLoaderClient:/webkit/WebKitWebPage/get-uri (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26521 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49523 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27851 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50105 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52621 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/157463 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51966 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39428 "Found 8 new JSC stress test failures: wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.default-wasm, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-bbq, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-bbq-no-consts, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-collect-continuously, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-eager, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-eager-jettison, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-no-cjit, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-slow-memory (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55314 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53663 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 reviewed (approved):
Seems fine since we don’t support MSVC anymore

rr-codes committed:
Fix the JavaScriptCore build in some configurations
https://bugs.webkit.org/show_bug.cgi?id=300635
rdar://162411683

Reviewed by Keith Miller.

First, some background:

1. The `asm` keyword is a supported keyword in C++, and GNU C. Notably however, it does not exist in non-GNU C standards.
Instead, the alternative keyword `__asm__` exists to serve the same purpose.

2. The bmalloc module map consists of a `pas` submodule, which participates in header fusion. This means that if a client
uses any of the headers in it, it implicitly imports all the other headers. This is expected and desired behavior.

3. The JavaScriptCore private module map defines a submodule "PASReportCrashPrivate" which contains the associated header,
and PASReportCrashPrivate.h transitively depends on at least one of the headers in the `bmalloc.pas` modules.

4. Within the bmalloc.pas module is pas_utils.h and pas_utils_prefix.h, both of which use the `asm` keyword.

As a result of all of these, the JavaScriptCore private module map depends on the existence of the `asm` keyword in all cases,
whereas before it was limited to the inclusion of specific headers. Consequently, any non-GNU C clients of the JavaScriptCore private
module will fail to compile because `asm` does not exist.

Fix by moving to `__asm__` which is available everywhere.

* Source/bmalloc/libpas/src/libpas/pas_utils.h:
(pas_assertion_failed):
(pas_reverse):
(pas_reverse64):
(pas_atomic_store_uint8):
(pas_compare_and_swap_uint8_weak):
(pas_compare_and_swap_uint8_strong):
(pas_compare_and_swap_uint16_weak):
(pas_compare_and_swap_uint32_weak):
(pas_compare_and_swap_uint32_strong):
(pas_compare_and_swap_uint64_weak):
(pas_compare_and_swap_uint64_strong):
(pas_store_store_fence):
(pas_opaque):
(pas_compare_and_swap_pair_weak):
(pas_compare_and_swap_pair_strong):
(pas_atomic_store_pair):
(pas_compare_ptr_opaque):
* Source/bmalloc/libpas/src/libpas/pas_utils_prefix.h:
(__pas_compiler_fence):
(__pas_fence):
(__pas_depend_impl):

Canonical link: https://commits.webkit.org/301429@main
